### ISSUE TITLE
Deep Sea: Nullify Threat Outro Cutscene Doors Not Fully Closing

### ISSUE THREAD

**Issue by PeeMoldy** [fgsfdsfgs/perfect_dark#291]
So in the PC Port, when you beat the level Deep Sea: Nullify Threat, the outro cutscene shows the Cetan ship blowing up(also shown if you fail and don't escape in time), but the final door seen at the very end of the cutscene doesn't fully close. I'm not sure if the 2nd door fully closes either, but the one you see in the area with the purple suited shotgun cloaking guards that's before the 4 guards where you can get the Proximity Mines secret most certainly doesn't close during the cutscene. Can this be patched with the new patching system?

**Comment by RyanDwyer** [comment]
It fully closes when played from the cinema menu, but not when you play the level. It turns out Jo's geometry is blocking the door from closing even though she isn't rendered. You can tell this because you can change the close amount based on where you stand in the doorway.

On N64, it takes a while to load the cutscene animation on the final frame. During this time Jo's momentum carries her clear of the doorway. On the port, the cutscene loads within the same frame so she doesn't clear the doorway.

You can block the door on N64 too if you move through it slowly or at an angle.

**Comment by fgsfdsfgs** [comment]
Thanks for the explanation. In that case, I think this is not really worth fixing, especially since this can happen on the N64 as well. In fact I'm not even sure how you would fix this, other than maybe injecting extra commands into the cutscene list that teleport Jo away or something.

**Comment by PeeMoldy** [comment]
> Thanks for the explanation. In that case, I think this is not really worth fixing, especially since this can happen on the N64 as well. In fact I'm not even sure how you would fix this, other than maybe injecting extra commands into the cutscene list that teleport Jo away or something.

Maybe worth a try?

**Comment by LanHikariDS** [comment]
> Thanks for the explanation. In that case, I think this is not really worth fixing, especially since this can happen on the N64 as well. In fact I'm not even sure how you would fix this, other than maybe injecting extra commands into the cutscene list that teleport Jo away or something.

That or delay the cutscene start by maybe a quarter (if not half) of a second, to get an end result closer to original hardware

**Comment by fgsfdsfgs** [comment]
That would work as well, but I don't know how to do that without dirty hacks.

**Comment by fgsfdsfgs** [comment]
No, it's just that it would be a very ugly way to fix this one extremely specific issue and I'd want to avoid adding any more of those into the code.

**Comment by PeeMoldy** [comment]
> No, it's just that it would be a very ugly way to fix this one extremely specific issue and I'd want to avoid adding any more of those into the code.

Maybe the fix could be made togglable Ship of Harkinian style?

**Comment by LanHikariDS** [comment]
> > No, it's just that it would be a very ugly way to fix this one extremely specific issue and I'd want to avoid adding any more of those into the code.
> 
> Maybe the fix could be made togglable Ship of Harkinian style?

Eh, I can see the aversion. Even if it were togglable, that's not a great precedent, as we don't want to end up with a hundred toggles for every incredibly minor thing
